src/map/status.cpp

```diff
@@ -12144,9 +12144,9 @@ int32 status_change_start(struct block_list* src, struct block_list* bl,enum sc_
 			}
 			break;
 		case SC_TUNAPARTY:
-			val2 = (status->max_hp * (val1 * 10) / 100); // Max HP% to absorb
-			if (sd && pc_checkskill(sd, SU_SPIRITOFSEA))
-				val2 *= 2; // Double the shield life
+			val2 = status_get_max_hp(src) * (val1 * 10) / 100; // Shield according to the Caster's MaxHP%
+			if (pc_checkskill(reinterpret_cast<map_session_data*>(src), SU_SPIRITOFSEA) > 0)
```

Comment:
```suggestion
			if (map_session_data* ssd = BL_CAST(BL_PC, src); pc_checkskill(ssd, SU_SPIRITOFSEA) > 0)
```
Here you cannot use reinterpret_cast, since you cannot be sure that src is always a player (unless I missed something above).

Note: pc_checkskill contains a nullptr check.